src/devices/bus/megadrive/paprium.h

```diff
@@ -0,0 +1,252 @@
+#define ppm_bgm_addr(num) (ppm_bgm_tracks_base_addr + swapshorts(ppm_bgm_tracks_offsets[(num)]))
+
+// dual port ram data
+typedef struct ppm_sat_item_struct
+{ // VDP specs
+	uint16_t posY;
+	uint16_t sizeNext;
+	uint16_t attrs;
+	uint16_t posX;
+} ppm_sat_item_struct;
+
+typedef struct ppm_intf_object_struct
+{
+	uint16_t anim;
+	uint16_t nextAnim;
+	uint16_t objID; // b15 = fresh assign ?
+	uint16_t field_6;
+	uint16_t attrs;
+	uint16_t animCounter;
+	int16_t posX;
+	int16_t posY;
+} ppm_intf_object_struct;
+
+typedef struct ppm_dma_command_struct
+{
+	uint16_t autoinc;
+	uint16_t lenH;
+	uint16_t lenL;
+	uint16_t srcH;
+	uint16_t srcM;
+	uint16_t srcL;
+	uint16_t cmdH;
+	uint16_t cmdL;
+} ppm_dma_command_struct;
+
+typedef struct ppm_interface_struct
+{
+	uint16_t vectors[0x80];	   // 68k vectors table
+	uint16_t rom_header[0x80]; // megadrive header data
+	uint16_t scaling_buffer[0x300];
+	uint16_t save_buffer[0x100]; // also part of scaling data
+	uint16_t _padding[0x80];
+	ppm_sat_item_struct sat_data[144];   // sprite allocation table sits here
+	ppm_intf_object_struct obj_data[64]; // objects table
+	uint16_t _padding2[0x40];
+	// 0x1400
+	ppm_dma_command_struct dma_commands[121];
+	uint16_t audio_data[0x70 / 2];
+	uint16_t network_data[0x108];
+	union
+	{
+		uint16_t command_args[128];
+		uint32_t command_args_long[64];
+	};
+	uint16_t dma_total;	// total size in words
+	uint16_t dma_budget;	// per frame budget (depends on system)
+	uint16_t dma_remaining; // remaining budget for current frame
+	uint16_t dma_commands_count;
+	uint16_t sat_count; // # of items present in SAT table
+	uint16_t unknown_1f1a;
+	uint16_t unknown_1f1c;
+	uint16_t unknown_1f1e;
+	uint16_t buffer[0x60];
+	// main registers
+	uint16_t reg_unk0;     // 1fe0
+	uint16_t reg_unk1;     // 1fe2
+	uint16_t reg_status_1; // 1fe4
+	struct
+	{
+		union
+		{
+			uint16_t word;
+			struct
+			{
+				uint16_t megawire_status : 3;
+				uint16_t bit_3 : 1;
+				// sub_9B738
+				uint16_t bit_4 : 1;	       // MW? send a900 if set
+				uint16_t megawire_data_in : 1; // MW ack ?
+				uint16_t bit_6 : 1;
+				uint16_t bit_7 : 1;
+
+				uint16_t eeprom_error1 : 1; // b8 sum error?
+				uint16_t eeprom_error2 : 1; // b9 eepr error?
+				uint16_t bit_10 : 1;
+				uint16_t bit_11 : 1;
+
+				uint16_t bit_12 : 1;
+				uint16_t bit_13 : 1;
+				uint16_t busy : 1; // b14 task busy
+				uint16_t bit_15 : 1;
+			} bits;
+		};
+	} reg_status_2;	   // 1fe6
+	uint16_t reg_unk4; // 1fe8
+	uint16_t reg_command;
+	uint16_t reg_watchdog; // unconfirmed
+	uint16_t reg_unk7;     // 1fee
+	uint16_t reg_unk8;     // 1ff0
+	uint16_t reg_unk9;     // 1ff2
+	uint16_t reg_unkA;     // 1ff4
+	uint16_t reg_unkB;     // 1ff6
+	uint16_t reg_unkC;     // 1ff8
+	uint16_t reg_unkD;     // 1ffa
+	uint16_t reg_unkE;     // 1ffc
+	uint16_t reg_unkF;     // 1ffe
+} ppm_interface_struct;
+
+// sdram data
+#define PPM_DATA_START_ADDR 0x10000 // base sdram unpack address
```

Comment:
Is there a need for this define? It looks local to `ppm_setup_data` function only.